src/runtime/mini-prompt-api.js

```diff
@@ -0,0 +1,240 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AnalyticsEvent} from '../proto/api_messages';
+import {AutoPromptType} from '../api/basic-subscriptions';
+import {assert} from '../utils/log';
+import {createElement} from '../utils/dom';
+import {msg} from '../utils/i18n';
+import {setStyle} from '../utils/style';
+
+/** @type {!Object<string, string>} */
+const SUBSCRIPTION_TITLE_LANG_MAP = {
+  'en': 'Subscribe with Google',
+  'ar': 'Google اشترك مع',
+  'de': 'Abonnieren mit Google',
+  'es': 'Suscríbete con Google',
+  'es-latam': 'Suscríbete con Google',
+  'es-latn': 'Suscríbete con Google',
+  'fr': "S'abonner avec Google",
+  'hi': 'Google के ज़रिये सदस्यता',
+  'id': 'Berlangganan dengan Google',
+  'it': 'Abbonati con Google',
+  'jp': 'Google で購読',
+  'ko': 'Google 을 통한구독',
+  'ms': 'Langgan dengan Google',
+  'nl': 'Abonneren via Google',
+  'no': 'Abonner med Google',
+  'pl': 'Subskrybuj z Google',
+  'pt': 'Subscrever com o Google',
+  'pt-br': 'Assine com o Google',
+  'ru': 'Подпиcka через Google',
+  'se': 'Prenumerera med Google',
+  'th': 'สมัครฟาน Google',
+  'tr': 'Google ile Abone Ol',
+  'uk': 'Підписатися через Google',
+  'zh-tw': '透過 Google 訂閱',
+};
+
+// TODO(stellachui): Add translated contribution strings here or from strings.js
+//   when we get them.
+/** @type {!Object<string, string>} */
+const CONTRIBUTION_TITLE_LANG_MAP = {
+  'en': 'Contribute with Google',
+};
+
+const TITLE_CONTAINER_DIV_HTML = `
+<div class="swg-mini-prompt-icon-$theme$"></div>
+<h1 class="swg-mini-prompt-title-text-$theme$">$textContent$</h1>
+`;
+
+const CLOSE_CONTAINER_DIV_HTML = `
+<div class="swg-mini-prompt-close-icon-$theme$"></div>
+`;
+
+/**
+ * Class for handling the display (and logging) of the mini prompt.
+ */
+export class MiniPromptApi {
+  /**
+   * @param {!./deps.DepsDef} deps
+   */
+  constructor(deps) {
+    /** @private @const {!../model/doc.Doc} */
+    this.doc_ = deps.doc();
+
+    /** @private @const {?./client-config-manager.ClientConfigManager} */
+    this.clientConfigManager_ = deps.clientConfigManager();
+    assert(
+      this.clientConfigManager_,
+      'MiniPromptApi requires an instance of ClientConfigManager.'
+    );
+
+    /** @private @const {!./client-event-manager.ClientEventManager} */
+    this.eventManager_ = deps.eventManager();
+  }
+
+  /**
+   * Does the setup required for the mini prompt's display, including injecting
+   * the mini prompt's css.
+   */
+  init() {
+    const head = this.doc_.getHead();
+    if (!head) {
+      return;
```

Comment:
Could you add a warning message and also a test for this?

https://github.com/subscriptions-project/swg-js/blob/42fa085c6de03cced89bcf298b09729ce6017ec4/src/api/entitlements.js#L181-L183

Reply:
Added.